### PR TITLE
Updating docs for testing.rst content

### PR DESCRIPTION
**Pull request by downzer0**
This brings over old content from `testing.rst` into a single place for developers.
## Tasks
- [x] Bring over testing.rst content
- [x] Include i18n_dummy content
## Reviewers
- [ ] @catong 
- [ ] @clytwynec 
- [ ] @jzoldak 
- [ ] @benpatterson

### CODE REVIEW

**Comment by jzoldak** [comment]
Will there be another PR that removes testing.rst from the platform (or replaces it with a link to ReadTheDocs)?

One concern that I've heard about removing docs from the platform repo is that if you are developing offline (or with a poor internet connection), it's nicer to know that you have everything you need offline and in one place. Are we recommending that developers also clone this repo?

**Comment by downzer0** [comment]
It sounded like Read the Docs was the primary go-to source, so I think adding a link from testing.rst to it would be useful. I don't think it would be wise to have this information in two different places (maintainability) so whichever we decide to keep, we should make sure everyone is in sync.

File: en_us/developers/source/testing/code-quality.rst [hunk 1]
```diff
@@ -20,10 +23,12 @@ in the process of introducing other changes.
 To check the quality of your pull request, just go to the top level of the
 edx-platform codebase and run::
 
-    $ paver run_quality
+    ``$ paver run_quality``
+
+You can also use the following commands individually to run each set of tests respectively.
```
Comment (clytwynec) [inline en_us/developers/source/testing/code-quality.rst:28]: I'd substitute 'linter' for 'set of tests'.

File: en_us/developers/source/testing/code-quality.rst [hunk 2]
```diff
@@ -32,15 +37,43 @@ violations are just too challenging to resolve, so we opt to ignore them via
...
 in the line.
 
+JavaScript
+----------
+
+To view JavaScript code style quality run this command.
+
+    ``paver run_jshint``
+
+This command also comes with a ``--limit`` switch, this is an example of that switch.
```
Comment (clytwynec) [inline en_us/developers/source/testing/code-quality.rst:54]: Might be useful to mention that using this flag means that the command will return a non-zero exit code if there are more violations than the specified limit.

File: en_us/developers/source/testing/code-quality.rst [hunk 3]
```diff
@@ -32,15 +37,43 @@ violations are just too challenging to resolve, so we opt to ignore them via
...
+
+    ``paver run_jshint --limit=700``
+
+Overall Complexity
+------------------
+
+Two tools are available for evaluating complexity of edx-platform code.
+
+* `radon`_ for Python code complexity.
+  * To obtain complexity, run
```
Comment (clytwynec) [inline en_us/developers/source/testing/code-quality.rst:64]: "To generate a complexity report for python code in `reports/complexity/python_complexity.log`, run"  maybe?

File: en_us/developers/source/testing/internationalization.rst [hunk 4]
```diff
@@ -0,0 +1,19 @@
+********************
+Internationalization
+********************
+
+Any new text that is added should be internationalized and translated.
+
+Testing translations with dummy translations
+============================================
+
+Any text you add to the platform should be internationalized. To generate translations for
```
Comment (clytwynec) [inline en_us/developers/source/testing/internationalization.rst:10]: Not sure if it's needed or even accurate, but should this specify "user-facing text"?

File: en_us/developers/source/testing/running-tests.rst [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+
+You can run the entire suite of unit-level tests at once by running the command.
+
+    ``paver test``
+
+This includes Python, JavaScript, and documentation tests. It does not, however,
+run any acceptance tests.
+
+**Note:** paver is a scripting tool. To get information about various options,
+you can run the this command.
```
Comment (clytwynec) [inline en_us/developers/source/testing/running-tests.rst:19]: Typo: "run _the this_ command."

**Comment by clytwynec** [comment]
I noticed the whole thing is rewritten and some of the information is omitted.  Why not just copy over the existing testing.rst and fix confusing or missing parts?  The two big things I noticed are missing are the intro section with the testing pyramid, and the section about options for speeding up bok choy test development.

**Comment by downzer0** [comment]
@clytwynec Thanks for the initial feedback. I re-wrote most of it because there's a specific format required for the documentation files and copying/pasting wasn't translating well. I found it easier to just retype things, despite the few typos which I'll address.

I think I moved the "speed up/optimize" stuff to a different section, thinking that this portion was for when developers run testing locally. But now that you mention it, it might make more sense here instead.

**Comment by clytwynec** [comment]
All of the info in the existing testing.rst is meant for developers running tests in devstack.  Most of the part about speeding up the bok choy tests is actually _only_ for running them locally.

**Comment by benpatterson** [comment]
@clrux I think scope has creeped beyond the original intent of your change. Can we close this PR and just update the testing.rst with the translation info that you originally wanted to put into place?

I think the decision to move testing.rst content needs to be handled separately and warrants a larger discussion with docs.

**Comment by downzer0** [comment]
@benpatterson Scope has definitely crept. Happy to simplify this.

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @downzer0! I've created [OSPR-5336](https://openedx.atlassian.net/browse/OSPR-5336) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.


:warning: We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://open.edx.org/wp-content/uploads/2019/01/individual-contributor-agreement.pdf)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

**Comment by nedbat** [comment]
Hi, sorry for the new comment on this old pull request. We're testing out re-scanning contributions, and our bot added a confusing message that you can ignore. We'll fix that. Sorry about the noise.

**Pull request closed without merging**